### PR TITLE
Add twig extension and autoUrl filter

### PR DESCRIPTION
**Pull request by f-elix**
This PR adds a twig extension class so we can define global filters, functions, etc. 

As an example, I converted the autoUrl util function to a filter accessible from everywhere, meaning we can now do `{{ button|autoUrl }}`. Maybe it can be expanded? 

Side note, I had trouble when putting the module files inside a `src` folder like the other modules. The only way I could get it working properly was by putting the files directly in the folder, without a `src` folder.

### CODE REVIEW

File: modules/twigextensionmodule/TwigExtensionModule.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+use modules\twigextensionmodule\twigextension\TwigExtension;
+
+use Craft;
+
+use yii\base\Event;
+use yii\base\InvalidConfigException;
+use yii\base\Module;
+
+/**
```
Comment (fhamon) [inline modules/twigextensionmodule/TwigExtensionModule.php:17]: Please remove scaffolder's crappy comments.

File: config/app.php [hunk 2]
```diff
@@ -27,10 +27,14 @@
         ],
         'language-redirect' => [
             'class' => \modules\languageRedirect\LanguageRedirect::class,
+		],
+        'twig-extension' => [
+            'class' => \modules\twigextensionmodule\TwigExtensionModule::class,
```
Comment (fhamon) [inline config/app.php:32]: rename `twigextensionmodule` to twigextension please

Comment (fhamon) [reply config/app.php:32]: also `TwigExtensionModule` to `TwigExtension`

Comment (fhamon) [reply config/app.php:32]: don't forget to change all instences of them too !

**Review by fhamon — CHANGES_REQUESTED** [review]

File: modules/twigextensionmodule/twigextension/TwigExtension.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<?php
```
Comment (fhamon) [inline modules/twigextensionmodule/twigextension/TwigExtension.php:1]: Use spaces for indenting please (yeah I know. yuk.) we only use spaces in PHP bc of psr complience.

**Review by fhamon — CHANGES_REQUESTED** [review]

File: config/app.php (hunk 2 above)
Comment (f-elix) [reply config/app.php:32]: Won't it clash with the actual TwigExtension class?

Comment (fhamon) [reply config/app.php:32]: In therory it should not bc of the "use" statatments at the beginning of each php files.

Comment (fhamon) [reply config/app.php:32]: you can have multiple classes named the same way in php. it's kinda weird, but it works with the help of "use" https://www.php.net/manual/en/language.namespaces.importing.php

File: modules/twigextensionmodule/twigextension/TwigExtension.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ * @copyright Copyright (c) 2020 Deux Huit Huit
+ */
+
+namespace modules\twigextensionmodule\twigextension;
+
+use modules\twigextensionmodule\TwigExtensionModule;
+
+use Craft;
+
+use Twig_Extension;
```
Comment (fhamon) [inline modules/twigextensionmodule/twigextension/TwigExtension.php:14]: this is a good example if you have the same class name twice and you need to use both in the same file you can always "rename" one locally in the file with use.

e.g.: `use Twig_Extension as HelloWorld`

and Then create a class like so `class Twig_Extension extends HelloWorld` ...

Comment (f-elix) [reply modules/twigextensionmodule/twigextension/TwigExtension.php]: Good thinking! I did something similar in the module file.

**Review by fhamon — APPROVED** [review]

**Pull request merged**